### PR TITLE
docs(setUserId): Add documentation about `setUserId`

### PR DESCRIPTION
**Pull request by BrodaNoel**
### Summary
Adds documentation about `setUserId`

### Checklist

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-TypeScript/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  No

### CODE REVIEW

**Review by liuyang1520 — APPROVED** [review]

File: packages/analytics-react-native/README.md [hunk 1]
```diff
@@ -33,6 +33,17 @@ Initialization is necessary before any instrumentation is done. The API key for
 amplitude.init(API_KEY);
 ```
 
+### Set UserId
+
+Sets an `userId` (usually called after user logs in).
+``typescript
+import { setUserId } from '@amplitude/analytics-react-native';
+
+// Track a basic event
```
Comment (BeepYoung) [inline packages/analytics-react-native/README.md:42]: There is no tracking logic so maybe remove this comment?

Comment (BrodaNoel) [reply packages/analytics-react-native/README.md]: Done!

File: packages/analytics-react-native/README.md [hunk 2]
```diff
@@ -33,6 +33,16 @@ Initialization is necessary before any instrumentation is done. The API key for
 amplitude.init(API_KEY);
 ```
 
+### Set UserId
+
+Sets an `userId` (usually called after user logs in).
+``typescript
```
Comment (liuyang1520) [inline packages/analytics-react-native/README.md:39]: Sorry, found one missing "`", others are LGTM, thanks for adding this to doc!

Comment (BrodaNoel) [reply packages/analytics-react-native/README.md]: Done

**Review by kevinpagtakhan — APPROVED** [review]

**Pull request merged**